src/common/Widget.js

```diff
@@ -578,6 +579,29 @@
     };
 
     //  DOM/SVG Node Helpers  ---
+    
+    Widget.prototype.showBusy = function (show) {
+        if (show) {
+            //show the indicator
+        } else {
+            //hide the indicator
+        }
+    };
+
+    Widget.prototype.pushBusy = function () {
+        this._busyCounter++;
+        if (this._busyCounter > 0) {
+            this.showBusy(true);
+        }
+    };
+
+    Widget.prototype.popBusy = function () {
+        this._busyCounter--;
+        if (this._busyCounter <= 0) {
```

Comment:
The only reason I did this is if there is a mistake by the developer and busyCounter gets <0 the loader will get stuck on the screen.  With <= it will still be removed.  I've seen similar things happen in other projects I've worked. Let me know which is the lesser of the two evils.